### ISSUE TITLE
Support `npm publish` in `emulator-sources` mode

### ISSUE THREAD

**Issue by carvinlo** [smapiot/piral#448]
# New Feature Proposal

For more information, see the `CONTRIBUTING` guide.

## Description

Publish the `app-shell` to the npm registry in a simple way like `npm publish` in `emulator-sources` mode.

## Background

It is generally necessary to publish the `app-shell` to a private registry for other developers to install.

## Discussion

The [files field](https://github.com/carvinlo/piral/commit/a38e1599f8ef08e543a329e277a4ffde462a9fea) needs to be preserved.

**Comment by FlorianRappl** [comment]
Well, usually you publish the `tgz` that comes out of the emulator. `emulator-sources` is just everything before making it a tarball. So what you want does not make much sense.

Regarding preservation of the `files` field. I don't think it's necessary and it should not be preserved. The "original" files field only makes sense on the top-level package.json. But all these references are not there in the generated package.json. Therefore, preserving the files field is actually an issue. Instead, all files are put into a dedicated files.tgz file, which preserves them.

Long story short: On this item I don't know why and for what. You can always use `npm publish`. Maybe you can describe the use case and why you think the currently available solution does not fit there.

**Comment by carvinlo** [comment]
All is well! 
It's just that publishing the `tgz` here requires complicated configuration.

**Comment by FlorianRappl** [comment]
> It's just that publishing the tgz here requires complicated configuration.

I see, good point. So you cannot just `npm publish foo.tgz`? However, the emulator-sources dir should be 100% the same as the content of the tgz from the emulator dir. If its not we should look into that.
